### PR TITLE
test: w3up-client tsconfig.json now includes "./test" so its tests have some type checking

### PR DESCRIPTION
**Pull request by gobengo**
Non-goal
* changing src code
* avoidance of all ts-ignore in these newly included files (that can come as iteration)

Context:
* https://github.com/web3-storage/w3protocol/issues/668#issuecomment-1485989307

### CODE REVIEW

File: packages/w3up-client/src/client.js
Comment (gobengo) [inline packages/w3up-client/src/client.js:127]: reverting this ts api interface change https://github.com/web3-storage/w3up-client/commit/ee1cf3a30a79a98c21f2e897d6a26e443b41390f#r106486919

(let's introduce the `DID` -> `DID<'key'>` again in one fell swoop that also adjusts some `@web3-storage/access/agent-data` types that still need to be done)

File: packages/w3up-client/test/capability/access.test.js
Comment (gobengo) [inline packages/w3up-client/test/capability/access.test.js:36]: I may be able to get rid of these ts-ignore (remember: the whole file was ignored by tsc before) in a followup PR by removing the `voucher` type prop from `@web3-storage/access/types` ServiceType.

**Review by travis — APPROVED** [review]
this looks good to me! I'd definitely feel a bit comfier if @alanshaw or @Gozala weighed in since some of this stuff makes changes to capabilities that don't intuitively make sense to me, but it's tests and seems like an improvement over not typechecking this stuff, so +1

**Comment by gobengo** [comment]
> since some of this stuff makes changes to capabilities that don't intuitively make sense to me

@travis it makes changes to some *tests* that have the word 'capability' in their file name, but notably it does not make any changes to the `@web3-storage/capabilities` package, i.e. the canonical capability parsers that sorta define the protocol.

with that said, no worries at all waiting for a review from @alanshaw or @Gozala as well

File: packages/w3up-client/test/capability/space.test.js
Comment (alanshaw) [inline packages/w3up-client/test/capability/space.test.js:56]: Why these removals?

**Review by alanshaw — APPROVED** [review]

File: packages/w3up-client/src/client.js
Comment (gobengo) [reply packages/w3up-client/src/client.js:127]: oops this was not 'releasable' due to using `test:` conventional commit

Comment (gobengo) [reply packages/w3up-client/src/client.js:127]: This change went out in w3up-client@5.5.0

**Pull request merged**
